examples/polynomial/run_demo

```diff
@@ -1,7 +1,8 @@
 #!/usr/bin/env bash
 set -e
 
-ert3 init
+ert3 init --example polynomial
```

Comment:
This is a bit odd, as now we create a new workspace which is a copy of what was the intended workspace within the workspace 🤷 If we go down this route, which is probably a nice idea, we should not change the run_demo script, but the CI-pipeline. So that instead of it moving into the correct workspace and run poly_demo it does:

```
ert3 init --example polynomial (or spe1)
pushd polynomial
./run_demo
popd
```
and then we remove the init command from the `run_demo` script.

Reply:
Only weird if you run the script within the workspace, which you do not have to do I guess. Perhaps moving the run_demo script to e.g. tests/ert3demos/run_polynomial_demo would make it clear that it is supposed to work from anywhere.

Reply:
Sure, but then it is no longer a demo script... Currently, the demo script is part of the example in the sense that you can run it and get some results and inspect them afterwards. It also serves as an example and guidance for how one can interact with an ert3 workspace. We can move it into `tests/` and rename it to `test_polynomial`, but then we also need to accept that the examples does not come with a demo script. For which I'm fine with if that is your argument ;)

Reply:
Mostly wanted to point out that it doesn't need to be run from the examples folder. I think having a demo makes sense.

Reply:
Sure, but that is where it currently resides and then it is a bit odd that it recreates the data (and a new demo script) within the workspace again :)

Reply:
I agree. Another argument in favor of your position is that a regular user, in order to run the demo, would presumably have to run init simply to get the demo script anyways, so the init as part of the demo is just confusing.